main.go

```diff
@@ -50,6 +50,7 @@ var (
 	configFile     = kingpin.Flag("config.file", "Blackbox exporter configuration file.").Default("blackbox.yml").String()
 	timeoutOffset  = kingpin.Flag("timeout-offset", "Offset to subtract from timeout in seconds.").Default("0.5").Float64()
 	configCheck    = kingpin.Flag("config.check", "If true validate the config file and then exit.").Default().Bool()
+	dynamicProbe   = kingpin.Flag("enable.dynamic-probe", "If true, probes can be dynamically configured via GET parameter using /probe/dynamic endpoint.").Default().Bool()
```

Comment:
I'm also open to increase the security of this feature.

I was thinking about an addition flag like `dynamic-probe.authorized-clients` which defaults to 127.0.0.1. 

If someone really wants to have this feature on a public interface, it requires 2 steps (`enable` + `authorized-clients`)